src/server/dfly_bench.cc

```diff
@@ -966,7 +976,12 @@ int main(int argc, char* argv[]) {
 
   ClusterSpec shards;
   if (protocol == RESP) {
-    shards = proactor->Await([&] { return FetchCluster(ep, proactor); });
+    shards = proactor->Await([&] {
+      unique_ptr<FiberSocketBase, FiberSocketClose> socket(proactor->CreateSocket());
```

Comment:
+1 for `FiberSocketClose`. I am leaving this here for info, we also have `absl::Cleanup` which will perform a cleanup upon destruction. So `absl::Cleanup raii([&socket] { socket->close(); })` and because the order of destructors is reversed and the construction of cleanup comes after the construction of the uring socket we close at the end without passing this FiberSocketClose functor. 

The only reason I comment here, is so you can be aware that there is this utlity function which can provide this dirty RAII. 

p.s. FiberSocketClose is perfectly valid here so zero issues with that

Reply:
Thanks for info about `absl::Cleanup` i'll switch to cleanup with it since it is used through the code.

Reply:
I personally don't have any preference or care. It's quick and it works. Use whatever you like :smile: